Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,440 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+
+@[simp] lemma reflexivePair_obj_one : (reflexivePair f g s sl sr).obj one = A := rfl
+
+@[simp] lemma reflexivePair_map_right : (reflexivePair f g s sl sr).map .left = f := rfl
+
+@[simp] lemma reflexivePair_map_left : (reflexivePair f g s sl sr).map .right = g := rfl
+
+@[simp] lemma reflexivePair_map_reflexion : (reflexivePair f g s sl sr).map .reflexion = s := rfl
+
+end
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingReflexivePair ⥤ C := reflexivePair f g (commonSection f g)
+
+/-- The natural isomorphism between the diagram obtained by forgetting the reflexion of
+`ofIsReflexivePair f g` and the original parallel pair. -/
+noncomputable def inclusionWalkingReflexivePairOfIsReflexivePairIso
+    (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingParallelPair.inclusionWalkingReflexivePair ⋙ (ofIsReflexivePair f g) ≅
+      parallelPair f g :=
+  diagramIsoParallelPair _
+
+end Limits
+
+namespace Limits
+
+variable {C : Type u} [Category.{v} C]
+
+namespace reflexivePair
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+section
+section NatTrans
+
+/-- A constructor for natural transforms to a diagram of the form `reflexivePair f g s`. -/
+def mkNatTrans {F G : WalkingReflexivePair ⥤ C}
+    (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    (h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left := by aesop_cat)
+    (h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right := by aesop_cat)
+    (h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion := by aesop_cat) :
+    F ⟶ G where
+  app := fun x ↦ match x with
+    | zero => e₀
+    | one => e₁
+  naturality _ _ f := by
+    cases f
+    all_goals
+      dsimp
+      simp only [Functor.map_id, Category.id_comp, Category.comp_id,
+        Functor.map_comp, h₁, h₂, h₃, reassoc_of% h₁, reassoc_of% h₂,
+        reflexivePair_map_reflexion, reflexivePair_map_left, reflexivePair_map_right,
+        Category.assoc]
+
+@[simp]
+lemma mkNatTrans_app_zero {F G : WalkingReflexivePair ⥤ C}
+    (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    (h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left)
+    (h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right)
+    (h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion) :
+    (mkNatTrans e₀ e₁ h₁ h₂ h₃).app zero = e₀ := rfl
+
+@[simp]
+lemma mkNatTrans_app_one {F G : WalkingReflexivePair ⥤ C}
+    (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    (h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left)
+    (h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right)
+    (h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion) :
+    (mkNatTrans e₀ e₁ h₁ h₂ h₃).app one = e₁ := rfl
+
+
+variable {F G: WalkingReflexivePair ⥤ C}
+
+/-- Constructor for natural isomorphisms with a `reflexivePair`. -/
+@[simps!]
+def mkNatIso (u : F.obj one ≅ G.obj one) (v : F.obj zero ≅ G.obj zero)
```

Comment:
It would be better to use the same order of parameters as for `mkNatTrans`. (The proof could be exactly the same, or by constructing `hom`, and `inv` using `mkNatTrans` as I suggested.)
```suggestion
def mkNatIso (e₀ : F.obj zero ≅ G.obj zero) (e₁ : F.obj one ≅ G.obj one)
```

Reply:
(Same for `whiskerRightMkNatTrans`.)